api/v1alpha1/shared_types.go

```diff
@@ -63,6 +63,25 @@ const (
 	SecretTypeKuadrantCoreDNS corev1.SecretType = "kuadrant.io/coredns"
 
 	DefaultProviderSecretLabel = "kuadrant.io/default-provider"
+
+	// This label defines the GVR to use for the endpoints provider
+	EndpointGVRKey = "ENDPOINT_GVR"
+
+	// The GVR to use in the endpoint provider is none are provided
+	DefaultEndpointGVR = "kuadrant.io/v1alpha1.dnsrecords"
+
+	// This is a label selector for the endpoint provider to find available zones
+	EndpointLabelSelectorKey = "ENDPOINT_ZONE_RECORD_LABEL"
+
+	// Default label selector used by the endpoint provider
+	DefaultLabelSelector = "kuadrant.io/kuadrant_zone_record=true"
```

Comment:
While I'm not sure I've seen any hard rules about label names, I've always followed the convention of using hyphens over underscores in them since that seemed to be what [kubernetes](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/) themselves did. Either way, we should probably decide what we want to do and stick to a convention, I'd vote for hyphens for the reason above, it appears to be more common.

Also, i don't think this needs "kuadrant in it twice", would this not do:
```
kuadrant.io/zone-record=true
```

When we  are done we also need to make sure we go back to the rfc and diagrams and update to reflect reality.